src/ClusterManager/joblog_manager.py

```diff
@@ -37,51 +34,42 @@ def create_log(logdir='/var/log/dlworkspace'):
         logging.config.dictConfig(logging_config)
 
 
+_use_legacy = config.get('logging') not in [
+    'azure_blob', 'elaticsearch'
+] or config.get('__extract_job_log_legacy', False)
+
+
 def extract_job_log(job_id, log_path, user_id):
-    if elasticsearch_deployed and not config.get('__extract_job_log_legacy',
-                                                 False):
-        extract_job_log_with_elastic_search(job_id, log_path, user_id)
+    if not _use_legacy:
+        _extract_job_log(job_id, log_path, user_id)
     else:
-        extract_job_log_legacy(job_id, log_path, user_id)
+        _extract_job_log_legacy(job_id, log_path, user_id)
 
 
 @record
-def extract_job_log_with_elastic_search(jobId, logPath, userId):
+def _extract_job_log(jobId, logPath, userId):
     dataHandler = None
     try:
         dataHandler = DataHandler()
 
         old_cursor = dataHandler.GetJobTextField(jobId, "jobLogCursor")
         if old_cursor is not None and len(old_cursor) == 0:
             old_cursor = None
-        (logs, new_cursor) = GetJobLog(jobId, cursor=old_cursor)
-
-        container_logs = {}
-        for log in logs:
-            try:
-                container_id = log["_source"]["docker"]["container_id"]
-                log_text = log["_source"]["log"]
-                if container_id in container_logs:
-                    container_logs[container_id] += log_text
-                else:
-                    container_logs[container_id] = log_text
-            except Exception:
-                logging.exception(
-                    "Failed to parse elasticsearch log: {}".format(log))
+        (pod_logs, new_cursor) = GetJobLog(jobId, cursor=old_cursor)
 
         jobLogDir = os.path.dirname(logPath)
         if not os.path.exists(jobLogDir):
             mkdirsAsUser(jobLogDir, userId)
 
-        for (container_id, log_text) in container_logs.items():
+        for (pod_name, log_text) in pod_logs.items():
             try:
-                containerLogPath = os.path.join(
-                    jobLogDir, "log-conatainer-" + container_id + ".txt")
-                with open(containerLogPath, 'a') as f:
+                podLogPath = os.path.join(
+                    jobLogDir, "log-pod-" + pod_name + ".txt")
+                with open(podLogPath, 'a') as f:
```

Comment:
Can we add `encoding="utf-8"` for safety?

Reply:
Will rebase

Reply:
Rebased